### PR TITLE
adding voltage check calculation and final export as zip file

### PR DESCRIPTION
**Pull request by schwitzguebel**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/DichotomyParallelization.java [hunk 1]
```diff
@@ -35,31 +40,35 @@ public DichotomyParallelization(DichotomyLogging dichotomyLogging, DichotomyRunn
...
-        SweDichotomyResult sweDichotomyResult = new SweDichotomyResult(buildReturnedMap(direction, dichotomyResult));
-        return sweDichotomyResult;
+        return new SweDichotomyResult(direction, dichotomyResult, voltageMonitoringResult);
     }
 
-    private EnumMap<DichotomyDirection, DichotomyResult<RaoResponse>> buildReturnedMap(DichotomyDirection direction, DichotomyResult<RaoResponse> dichotomyResult) {
-        EnumMap<DichotomyDirection, DichotomyResult<RaoResponse>> mapResult = new EnumMap<>(DichotomyDirection.class);
-        mapResult.put(direction, dichotomyResult);
-        return mapResult;
+    private Optional<VoltageMonitoringResult> runVoltageCheck(SweData sweData, DichotomyResult<RaoResponse> dichotomyResult, DichotomyDirection direction) {
```
Comment (TheoPascoli) [inline gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/DichotomyParallelization.java:65]: Can you move out this method ? I dont think that it belongs to this class

File: gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/services/FileExporter.java [hunk 2]
```diff
@@ -94,6 +100,15 @@ public String saveVoltageMonitoringResultInJson(VoltageMonitoringResult result,
         return minioAdapter.generatePreSignedUrl(voltageResultPath);
     }
 
+    private String zipTargetNameToJsonName(String targetName) {
+        if (StringUtils.isNotBlank(targetName) &&
+                (targetName.contains(".ZIP") || targetName.contains(".zip"))) {
```
Comment (TheoPascoli) [inline gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/services/FileExporter.java:105]: can you replace those 2 conditions with "targetName.toLowercase().contains(".zip")" or it wont work ?

File: gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/services/OutputService.java [hunk 3]
```diff
@@ -25,9 +32,37 @@ public OutputService(FileExporter fileExporter) {
...
         TtcDocument ttcDoc = new TtcDocument(result);
         InputStream inputStream = ttcDoc.buildTtcDocFile();
         return fileExporter.exportTtcDocument(sweData, inputStream);
     }
+
+    public String buildAndExportEsFrVoltageDoc(SweData sweData, ExecutionResult<SweDichotomyResult> result) {
+        return buildAndExportVoltageDoc(DichotomyDirection.ES_FR, sweData, result);
+    }
+
+    public String buildAndExportFrEsVoltageDoc(SweData sweData, ExecutionResult<SweDichotomyResult> result) {
```
Comment (TheoPascoli) [inline gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/services/OutputService.java:45]: Could you set the direction as an argument so you dont have to repeat the method ?

**Review by TheoPascoli — CHANGES_REQUESTED** [review]

File: gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/DichotomyParallelization.java (hunk 1 above)
Comment (schwitzguebel) [reply gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/DichotomyParallelization.java:65]: sure in a seperate sevice maybe ?

**Review by WalAmeni — APPROVED** [review]

**Review by TheoPascoli — APPROVED** [review]

**Pull request merged**